crates/ruff/src/rules/flake8_type_checking/rules/typing_only_runtime_import.rs

```diff
@@ -181,9 +181,20 @@ fn is_implicit_import(this: &Binding, that: &Binding) -> bool {
             }
             BindingKind::Importation(Importation {
                 full_name: that_name,
-                ..
-            })
-            | BindingKind::SubmoduleImportation(SubmoduleImportation {
+                name: that_alias,
+            }) => {
+                if that_name == that_alias {
+                    // Ex) `pkg` vs. `pkg`
+                    this_name == that_name
+                } else {
+                    // Submodule importation with an alias: `import pkg.A as B`
+                    this_name
+                        .split('.')
+                        .zip(that_name.split('.'))
+                        .all(|(this_part, that_part)| this_part == that_part)
```

Comment:
Could we _just_ use this condition, since it will yield true when `this_name == that_name`?

Reply:
Not really. `foo` is not the same as `foo.bar` but it is an implicit import. The `zip` will stop when it can't move further making it an ideal choice of comparison.

Reply:
Right, but why do we need the `if that_name == that_alias { this_name == that_name }` portion? The tests seem to pass without it.

Reply:
In other words: is this alone not sufficiently general to handle all cases?

```rust
BindingKind::Importation(Importation {
    full_name: that_name,
    ..
}) => {
    // Submodule importation with an alias: `import pkg.A as B`
    this_name
        .split('.')
        .zip(that_name.split('.'))
        .all(|(this_part, that_part)| this_part == that_part)
}
```

Reply:
Ah yes. That will work, I just thought to split it out to avoid doing this computation and take the fast path directly.